lib/Sema/SemaExpr.cpp

```diff
@@ -10866,7 +10888,19 @@ QualType Sema::CheckAddressOfOperand(ExprResult &OrigOp, SourceLocation OpLoc) {
 
   // In Checked C, the address-of operator always produces an unchecked pointer
   // type.
-  return Context.getPointerType(op->getType(), CheckedPointerKind::Unchecked);
+
+  // checked scope change types produced by unary address-of operator
```

Comment:
Here is some rewording for the comment:

// Checked scopes change the types the address-of (&) operator.  In a checked scope, the
// operator poduces an array_ptr<T>.  In an unchecked scope, it continues to produce T *.